### DISCUSSION TITLE
'Validation' for all properties of an object

### DISCUSSION THREAD

**Discussion by Chicagoan2016** [MarimerLLC/csla#4102] [Questions]
We are using business base as a Criteria object. Our requirement is simple: ensure at least one property is present and don't allow special characters.
I can get the list of properties for our object from 'FieldManager.GetRegisteredProperties();' and the business rule can have a constructor like ' public ValidateSearchCriteria(IEnumerable<IPropertyInfo> fields)'

but how could we retrieve the properties in the Execute method of the business rule? 

Kind Regards

**Comment by StefanOssendorf** [comment]
You can add them all as `InputProperties` to resolve their values in your rule. Or do I misunderstand something?

**Reply by Chicagoan2016** [reply]
@StefanOssendorf   you are right, I have to add them to InputProperties in the rule's constructor. I was hoping if there was a way to access those properties in the Execute without using something like
```cs
   var input = (string)context.InputPropertyValues[PrimaryProperty];
```
Essentially I want to create a 'generic' rule that could be used with any object.

Regards

**Reply by StefanOssendorf** [reply]
The question is how you determine "a property is present". You could for example cast to object and test for null. And for special cases like numbers to test for greater than zero. Or if the input is a string it must not contain ....
That's pretty much possible.

**Reply by rockfordlhotka** [reply]
If you create the rule as a private nested class inside the business class you can use `context.Target` to directly interact with the business object's properties - bypassing the `InputProperties` concept entirely.

This is the "old" or original way to create rules - before async rules existed. This is not safe to do in an async rule, but if you are creating a sync rule that is tightly coupled to the business type (hence the private nested type), then you can just use `Target`.
